File: back/api/Models/User/Address.cs
using System.ComponentModel.DataAnnotations;

namespace api.Models.User;

public class Address
{
  [Key] public Guid AddressId { get; init; }
  [Required] public int UserId { get; init; }
  [Required] public string Name { get; set; }
  [Required] public string PhoneNumber { get; set; }
  [Required] public string Email { get; set; }
  [Required] public string StreetAddress { get; set; }
  [Required] public string City { get; set; }
  [Required] public string State { get; set; }
  [Required] public string Zip { get; set; }
}

File: packages/database/src/types/users/Address.ts
import { DocumentStuff } from "..";

export interface IAddress {
  userId: string;
  name: string;
  phoneNumber: string;
  email: string;
  streetAddress: string;
  city: string;
  state: string;
  zip: string;
}

export type AddressDocument = IAddress & DocumentStuff;
